tljh/installer.py

```diff
@@ -133,28 +130,16 @@ def ensure_jupyterhub_service(prefix):
         jupyterhub_config_path=os.path.join(HERE, 'jupyterhub_config.py'),
         install_prefix=INSTALL_PREFIX,
     )
-    systemd.install_unit('configurable-http-proxy.service', proxy_unit_template.format(**unit_params))
```

Comment:
@minrk, I don't quite understand what you mean by leaving the configurable-http-proxy unit in place. Does that mean keep CHP as the default proxy and add an upgrade option somewhere in the installer that would stop CHP and use only traefik?
I've submitted a new commit addressing the other changes you suggested.

Reply:
Ah, I mean that after upgrading from a version of tljh with CHP, to this PR with only traefik, configurable-http-proxy is still running from the previous version, even though it's not being used anymore. It would be good to clean up after ourselves. We should add a `systemd.uninstall_unit` that removes and unregisters the configurable-http-proxy unit while we are upgrading.

Reply:
I've noticed there's already an ```uninstall_unit``` in ```systemd``` that we could use.

Please correct me if I got this wrong: basically, what we want is to keep ```configurable-http-proxy.service``` and maybe add to it an ```ExecStop``` so that when uninstall_unit gets called, CHP will be properly stopped?
Also, is it ok to make use of the flag ```initial_setup``` in ```bootstrap.py``` to identify an upgrade and based on this call ```systemd.uninstall_unit```?

Reply:
Ideally, I'd love for us to not have to differentiate between initial_setup and upgrades wherever possible. In the installer, the general way to do that is to just uninstall the unit if it exists, regardless of initial setup or upgrade. This keeps the install / upgrade process idempotent.